firmware/controllers/algo/engine2.cpp

```diff
@@ -119,14 +119,17 @@ void EngineState::periodicFastCallback() {
 	getLimpManager()->updateRevLimit(rpm);
 
 	// post-cranking fuel enrichment.
+	float m_postCrankingFactor = interpolate3d(
+		engineConfiguration->postCrankingFactor,
+		engineConfiguration->postCrankingCLTBins, Sensor::getOrZero(SensorType::Clt),
+		engineConfiguration->postCrankingDurationBins, engine->fuelComputer.running.timeSinceCrankingInSecs
+	);
 	// for compatibility reasons, apply only if the factor is greater than unity (only allow adding fuel)
-	if (engineConfiguration->postCrankingFactor > 1.0f) {
-		// use interpolation for correction taper
-		engine->fuelComputer.running.postCrankingFuelCorrection = interpolateClamped(0.0f, engineConfiguration->postCrankingFactor,
-			engineConfiguration->postCrankingDurationSec, 1.0f, engine->fuelComputer.running.timeSinceCrankingInSecs);
-	} else {
-		engine->fuelComputer.running.postCrankingFuelCorrection = 1.0f;
+	// if the engine run time is past the last duration bin, disable ASE in case the table is filled with values more than 1.0, helps with compatibility with older solutioin
+	if ((m_postCrankingFactor < 1.0f) || (engine->fuelComputer.running.timeSinceCrankingInSecs > engineConfiguration->postCrankingDurationBins[(sizeof(engineConfiguration->postCrankingDurationBins)/sizeof(engineConfiguration->postCrankingDurationBins[0]))-1])) {
```

Comment:
efi::size() or CRANKING_ADVANCE_CURVE_SIZE to improve readability please, and let's merge it!

Reply:
done!